system/service.py

```diff
@@ -1009,8 +1009,10 @@ def service_enable(self):
                 if rc != 0:
                     self.module.fail_json(msg="unable to set rcvar using sysrc", stdout=change_stdout, stderr=change_stderr)
 
-                # sysrc does not exit with code 1 on permission error => validate successful change using service(8)
-                rc, check_stdout, check_stderr = self.execute_command("%s %s %s" % (self.svc_cmd, self.name, "enabled"))
+                # sysrc does not exit with code 1 on permission error => validate successful change using rc.subr(8) functions
+                # Due to the behavior of some scripts, we couldn't use service(8) here.
+                # For example, "service sendmail enabled" always fail if sendmail_msp_queue_enable is set to NO.
+                rc, check_stdout, check_stderr = self.execute_command("sh -c '. /etc/rc.subr;name=%s;rcvar=%s;load_rc_config $name;run_rc_command %s'" % (self.name, self.rcconf_key, "enabled"))
```

Comment:
I've changed sh to /bin/sh. 

I think we can assume /bin/sh is always available at least current and earlier version of FreeBSD.